priceloop_api/priceloop/extras/table.py

```diff
@@ -0,0 +1,855 @@
+import json
+from enum import Enum
+from http import HTTPStatus
+from io import StringIO
+from typing import Any, Callable, List, Literal, Optional
+
+import attr
+import numpy as np
+import pandas as pd
+import requests
+from pandas.api.types import infer_dtype
+from retry import retry
+from retry.api import retry_call
+
+from priceloop_api.api.import_api import create_csv_import_job, get_csv_import_job
+from priceloop_api.api.table_api import (
+    create_table,
+    delete_table,
+    get_incomplete_table_query_data,
+    get_table,
+    get_table_data,
+    patch_table_data,
+    truncate_table,
+)
+from priceloop_api.client import AuthenticatedClient
+from priceloop_api.errors import UnexpectedStatus
+from priceloop_api.models import (
+    ApiColumn,
+    ApiColumnSchema,
+    ApiColumnWithNullableValues,
+    ApiColumnWithValues,
+    ApiOnConflict,
+    ApiTable,
+    ApiTableData,
+    ApiTableSchema,
+    Append,
+    Delete,
+    ExplicitExprType,
+    ImportJob,
+    Insert,
+    TableImportMode,
+    TableRow,
+    Update,
+)
+from priceloop_api.types import UNSET, Unset
+
+from .exceptions import (
+    ColumnMixedTypes,
+    ColumnsMismatch,
+    ColumnTypesMismatch,
+    PlatformError,
+    )
+
+LIMIT = 30000  # cells
+
+
+COLUMNS_TYPES_MAPPING = {
+    "int64": "number",
+    "object": "string",
+    "bool": "boolean",
+    "datetime64[ns]": "date",
+    "float64": "number",
+}
+
+PLATFORM_TO_PANDAS_MAPPING = {
+    "boolean": "bool",
+    "date": "datetime64[ns]",
+    "number": "float64",
+}
+
+
+class ColumnType(str, Enum):
+    Number = "number"
+    Text = "string"
+    Boolean = "boolean"
+    Date = "date"
+    Json = "json"
+    Formula = "formula"
+
+
+class ImportMode(str, Enum):
+    Append = "append"
+    Overwrite = "overwrite"
+
+class PatchMode(str, Enum):
+    Append = "append"
+    Delete = "delete"
+    InsertOnConflictIgnore = "insert_on_conflict_ignore"
+    InsertOnConflictFail= "insert_on_conflict_fail"
+    InsertOnConflictUpdate = "insert_on_conflict_update"
+    Update = "update"
```

Comment:
nice. Can we put those in model files, so we don't blow up this already big table.py file?